src/Pos/Lrc/Arbitrary.hs

```diff
@@ -0,0 +1,73 @@
+-- | Arbitrary instances for Lrc types.
+
+module Pos.Lrc.Arbitrary
+       ( InvalidRichmenStake (..)
+       , ValidRichmenStake (..)
+       ) where
+
+import           Universum
+
+import qualified Data.HashMap.Strict as HM
+import           Pos.Constants       (genesisMpcThd)
+import           Pos.Lrc.Types       (RichmenStake)
+import           Pos.Types.Address   (StakeholderId)
+import           Pos.Types.Core      (Coin, coinPortionDenominator, getCoinPortion,
+                                      mkCoin, unsafeGetCoin)
+import           Test.QuickCheck     (Arbitrary (..), Gen, choose)
+
+-- | Wrapper over 'RichmenStake'. Its 'Arbitrary' instance enforces that the stake
+-- distribution inside must be valid, i.e. all of the coins are non-zero and every
+-- stakeholder has sufficient coins to participate.
+newtype ValidRichmenStake = Valid
+    { getValid :: RichmenStake
+    } deriving (Show, Eq)
+
+instance Arbitrary ValidRichmenStake where
+    arbitrary = Valid <$> genRichmenStake
+
+-- | Wrapper over 'RichmenStake'. Its 'Arbitrary' instance enforces that the stake
+-- distribution inside must be invalid, i.e. one of the stakeholders does not have
+-- sufficient coins to participate.
+newtype InvalidRichmenStake = Invalid
+    { getInvalid :: RichmenStake
+    } deriving (Show, Eq)
+
+instance Arbitrary InvalidRichmenStake where
+    arbitrary = Invalid <$> do
+        validRichmenStake <- genRichmenStake
+        poorMan <- arbitrary
+        return $ HM.insert poorMan (mkCoin 0) validRichmenStake
```

Comment:
Precisely. I did this a few times in tests already - I'd say the odds of this happening are very low, assuming the generator won't have a preference (for stakeholder IDs already generated, for example).